### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by AshiqurRah**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java [hunk 1]
```diff
@@ -159,4 +159,24 @@ public void setTags(Set<Tag> tags) {
...
+            return false;
+        }
+
+        // check for same values in fields
+        MarkPersonDescriptor m = (MarkPersonDescriptor) other;
+        return getHomework().equals(m.getHomework())
+                && getHomeworkIndex().equals(m.getHomeworkIndex())
+                && getAttendance().equals(m.getAttendance())
+                && getAttendanceIndex().equals(m.getAttendanceIndex());
+    }
```
Comment (cfyjoanna) [inline src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java]: wouldn't it be ok for MarkPersonDescriptor to only keep HomeworkIndex and AttendanceIndex? doesn't seem like it needs to know about the description of the items?

File: src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java [hunk 2]
```diff
@@ -159,4 +159,24 @@ public void setTags(Set<Tag> tags) {
...
+        if (other == this) {
+            return true;
+        }
+
+        // checks for null object
+        if (!(other instanceof MarkPersonDescriptor)) {
+            return false;
+        }
+
+        // check for same values in fields
```
Comment (cfyjoanna) [inline src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java:87]: would be good to remove the other unnecessary fields from the class

File: src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java (hunk 1 above)
Comment (AshiqurRah) [reply src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java]: Did the necessary changes

File: src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java (hunk 2 above)
Comment (AshiqurRah) [reply src/main/java/seedu/address/logic/commands/MarkPersonDescriptor.java:87]: Can can, have made the changes

**Review by farisanadia — COMMENTED** [review]
looks good but maybe its better to have tests for somefieldsspecified and allfieldsspecified too... shld be similar to the remove ones but idk if those are correctly done either

**Review by farisanadia — APPROVED** [review]
LGTM

**Pull request merged**